src/containerapp/azext_containerapp/custom.py

```diff
@@ -3656,6 +3660,72 @@ def disable_dapr(cmd, name, resource_group_name, no_wait=False):
     except Exception as e:
         handle_raw_exception(e)
 
+def init_dapr_component(cmd, resource_group_name, environment_name):
+    _validate_subscription_registered(cmd, CONTAINER_APPS_RP)
+
+    # Create the Redis service
+    redis_capp_name = DAPR_REDIS_PREFIX +'-' + ''.join(random.choice(string.ascii_lowercase + string.digits) for _ in range(5))
+    logger.info("Creating Redis service '%s'...", redis_capp_name)
+    try:
+        redis_capp_def = create_redis_service(cmd, redis_capp_name, environment_name, resource_group_name)
+    except Exception as e:
+        raise ValidationError("Failed to create Redis service.") from e
+    
+    if redis_capp_def is None:
+        raise ValidationError("Failed to create Redis service, did not receive a response.")
+
+    # Look up the Redis secret
+    _get_existing_secrets(cmd, resource_group_name, redis_capp_name, redis_capp_def)
```

Comment:
@ahmelsayed Is there a better way to retrieve the Redis connection string from a service without binding it to another app?

Reply:
Sorry, what is this supposed to do here? you should be able to bind a redis dev service to a dapr component directly without needing to retrieve secrets.

Reply:
Can you share how to do it and which API version? Couldn't find service bind in the spec but maybe looking in the wrong place.

This is a command that performs the equivalent of `dapr init` in ACA.

Reply:
When you do a default `dapr init -k` for a standard cluster, it would use the k8s secretstore to reference the secret redis creates when it is provisioned. Since that's disabled, we'd need to fetch the secrets this way and either apply them directly to the component, or, go through the secret management layer that Container Apps provides if you don't want to touch them directly, though I think they'd still need to be fetched here.

Reply:
@ahmelsayed could you share more details here please?

Reply:
Making these changes from our offline discussion!